### PR TITLE
deps: Add build recipe for WebP

### PR DESCRIPTION
**Pull request by zachlewis**
As requested in #4387.

### CODE REVIEW

File: THIRD-PARTY.md [hunk 1]
```diff
@@ -60,6 +60,9 @@ SPDX-License-Identifier: BSD-3-Clause
   The test image in testsuite/cryptomatte/src/cryptoasset.exr is a small crop
   of a sample image from the Cryptomatte project.
 
+* WebP (c) Copyright 2010 Google Inc.
+  https://developers.google.com/speed/webp
+
```
Comment (lgritz) [inline THIRD-PARTY.md:65]: This isn't needed (not in THIRD-PARTY.md anyway) because no WebP code is vendored into our repository.

Comment (zachlewis) [reply THIRD-PARTY.md:65]: Ah, of course.

**Comment by lgritz** [comment]
I'm worried that the failure of the Windows CI tests is related to this somehow. DLL not in the right place, something like that?

**Comment by lgritz** [comment]
@zachlewis What's the current status of this one?

**Comment by zachlewis** [comment]
Help needed. I don't have a way of testing this locally, and I can't really tell what's going wrong from the logs.

**Comment by zachlewis** [comment]
Interesting, forcing WebP to only build static libraries seems to fix the Windows CI failures. Nice.

File: THIRD-PARTY.md [hunk 2]
```diff
@@ -307,4 +307,3 @@ Public domain
 -------------------------------------------------------------------------
 
 If we have left anything out, it is unintentional. Please let us know.
-
```
Comment (lgritz) [inline THIRD-PARTY.md:310]: I kinda wish PRs would not make unrelated formatting changes to other parts of the codebase, but whatever, this is harmless.

**Review by lgritz — APPROVED** [review]
LGTM

**Review by lgritz — APPROVED** [review]
LGTM
